### ISSUE TITLE
Hidden packages should be removed from author profile & related packages

### ISSUE THREAD

**Issue by domyen** [percolatestudio/atmosphere#268]
Packages hidden by `meteor admin set-unmigrated` should also be hidden from the author profile and related packages (on the package page) page.

Ref: https://github.com/percolatestudio/atmosphere-beta/issues/267#issuecomment-58772928

**Comment by tmeasday** [comment]
@dburles you can simple check to see if `package.latestVersion.unmigrated` is true and filter if so. We already do this in a few places as you'll see.

There's probably an edge case here if someone marks a new version of a working package as unmigrated, but this is actually not what unmigrated is intended for. Let's deal with that if it comes up.

**Comment by tmeasday** [comment]
Actually, on second thought, it could be worth doing a quick check to see if there's any package out there with an unmigrated latest version which has legitimate versions. @dburles can you ping me if you find some. It could be what @queso was talking about on meteor-core when he said there were missing packages.

**Comment by queso** [comment]
Was there some warning about this change?

**Comment by tmeasday** [comment]
@queso - I'm not sure what you mean?

There were a bunch of packages that failed to successfully migrate -- you will have noticed them on Atmosphere with "DOESN'T WORK IN 0.9.0" or something written on them.

As a part of 0.9.3, MDG added an `unmigrated` field to package versions and marked all the broken versions with that. They then hide the package from `meteor search` if it has no "non-unmigrated" versions (i.e. it's broken, no-one has fixed it). What we are doing here is doing the same in Atmosphere.

If we are hiding packages for which this isn't the case, then that's a mistake on our part. I'm trying to figure that out.

**Comment by udondan** [comment]
> There's probably an edge case here if someone marks a new version of a working package as unmigrated, but this is actually not what unmigrated is intended for. Let's deal with that if it comes up.

---

> Actually, on second thought, it could be worth doing a quick check to see if there's any package out there with an unmigrated latest version which has legitimate versions.

I yesterday marked one of my packages as unmigrated as it currently can not be fixed. When I did so, all versions have been marked:

```
$ meteor admin set-unmigrated udondan:contact
Setting udondan:contact@0.1.0 as unsuccessfully migrated ... done!
Setting udondan:contact@0.1.1 as unsuccessfully migrated ... done!
Setting udondan:contact@0.1.2 as unsuccessfully migrated ... done!
Setting udondan:contact@0.1.6 as unsuccessfully migrated ... done!
Setting udondan:contact@0.2.0 as unsuccessfully migrated ... done!
Setting udondan:contact@0.3.0 as unsuccessfully migrated ... done!
Setting udondan:contact@0.3.1 as unsuccessfully migrated ... done!
Setting udondan:contact@0.3.2 as unsuccessfully migrated ... done!
Setting udondan:contact@0.3.3 as unsuccessfully migrated ... done!
Setting udondan:contact@0.4.0 as unsuccessfully migrated ... done!
```

So it seems impossible to only mark the latest version as unmigrated and there should be no edge case. :)

**Comment by tmeasday** [comment]
@udondan - check `meteor admin help set-unmigrated`: 

```
Usage: meteor admin set-unmigrated <package name> [--success]
       meteor admin set-unmigrated <package name>@<version> [--success]

Set a version of a package, as unmigrated: make it invisible in search
and show, without a special option. (You should use this to hide packages that have been
renamed in the aftermath of the 0.9.0 migration.) This will not stop the users from
adding the package to their app.

If no version is passed in, all versions will be set. Use the --success option to
mark the package version as successfully migrated, making it show up again.

Options:
  --success         mark the package as successfully migrated.
```

So you can do a single version it seems (also you can undo what you did if it was an error ;) )

**Comment by udondan** [comment]
Double thanks! Didn't even know you can use the `help` command this way! Though the setting is ok as it is, since all versions won't work with 0.9. ;)

**Comment by tmeasday** [comment]
@dburles actually I created a separate ticket https://github.com/percolatestudio/atmosphere-beta/issues/270 -- So just assume that the data is correct, once that ticket is done it will be.
